components/Modal.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import FontAwesome from 'react-fontawesome';
+
+class ModalExample extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
```

Comment:
`toggle = () => { … }` to avoid `bind`

Reply:
wonderful - updated other functions in MapWithFilters.js too